app/liquid_tags/forem_tag.rb

```diff
@@ -0,0 +1,52 @@
+module ForemTag
+  REGISTRY_REGEXP = %r{#{URL.url}/\w+(/[\w-]+)?(/[\w-]+)?}
+  USER_ORG_REGEXP = %r{#{URL.url}/(?<name>[\w-]+)/?$}
+  POST_PODCAST_REGEXP = %r{#{URL.url}/(?<podcast>[\w-]+)/[\w-]+/?}
+  COMBINED_REGEXP = [USER_ORG_REGEXP, POST_PODCAST_REGEXP].freeze
+
+  def self.new(tag_name, input, parse_context)
+    link = process_input(input)
+    klass = determine_klass(link)
```

Comment:
This looks great!  I definitely want to see lots of tests of `determine_class` that's a lot of logic that is not quite verified.